### ISSUE TITLE
Adding support to use a resource without spec or status as a backing service

### ISSUE THREAD

**Issue by navidsh** [redhat-developer/service-binding-operator#389]
Currently, it is only possible to bind to a backing service that has `spec`/`status`. However, resources such as Secrets and ConfigMaps do not have `spec` and `status`.

This is to track the support following the discussion [here](https://github.com/redhat-developer/service-binding-operator/pull/367#issuecomment-596547734).

**Comment by sbose78** [comment]
Thanks!

**Comment by Avni-Sharma** [comment]
Currently, a workaround is there

That is to use mappings, initially known as customEnvVar.

Steps to reproduce:

Create a configMap:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: example
data:
  extra: "hello"
```
Create a sbr yaml as follows:
```
apiVersion: operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
  name: binding-request
spec:
  application:
    group: apps
    version: v1
    resource: deployments
    name: nodejs-rest-http-crud
  services:
  - group: ""
    version: v1
    kind: ConfigMap
    name: example
    id: s
  customEnvVar:
  - name: HELLO
    value: '{{ .s.data.extra }}'
```

now check the intermediate secret that is created. the secret that the sbr created will have data as HELLO: hello from the configmap

Will be adding an acceptance test for this

**Comment by pedjak** [comment]
Fixed by #984
